stablehlo/dialect/TypeInference.cpp

```diff
@@ -438,6 +439,11 @@ LogicalResult verifyReplicaGroups(std::optional<Location> location,
                              "groups cannot be empty");
 
   auto replicaIds = replicaGroups.getValues<int64_t>();
+
+  // all_reduce_c3
+  if (replicaGroups.getNumElements() == 0)
+    return emitOptionalError(location, "replica group cannot be empty");
```

Comment:
Where is this constraint coming from?

Reply:
This comes from the updated spec constraints. Updated the PR description to reflect it.

Reply:
I'm not sure I follow why we need to update the spec constraints in the first place.